exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillStatementImpl.java

```diff
@@ -204,7 +276,7 @@ public boolean isClosed() {
 
   @Override
   public int getMaxFieldSize() throws SQLException {
-    throwIfClosed();
+    throwIfTimedOutOrClosed();
```

Comment:
this methods cannot throw SqlTimeoutException... only `execute*` methods can, and eventually some ResultSet methods (but this part is optional per JDBC standard)